src/pybullet_industrial/interpolation.py

```diff
@@ -118,34 +138,73 @@ def circular_interpolation(start_point: np.array, end_point: np.array,
         planar_start_point, planar_end_point, radius, samples, clockwise)
 
-    path = np.zeros((3, samples))
+    positions = np.zeros((3, samples))
     for i in range(2):
-        path[all_axis[i]] = planar_path[i]
-    path[axis] = np.linspace(start_point[axis], end_point[axis], samples)
-    return ToolPath(path)
+        positions[all_axis[i]] = planar_path[i]
+    positions[axis] = np.linspace(start_point[axis], end_point[axis], samples)
+    final_path = ToolPath(positions=positions)
+
+    if start_orientation is not None and end_orientation is not None:
+        final_path.orientations = orientation_interpolation(
+            start_orientation, end_orientation, samples=samples)
 
+    return final_path
 
-def spline_interpolation(points: np.array, samples: int):
+
+def orientation_interpolation(start_orientation: np.array,
+                              end_orientation: np.array, samples: int):
+    """Interpolates between two orientations
+
+    Args:
+        start_orientation (np.array(3,1)): 3-D array containing the start orientation
+        end_orientation (np.array(3,1)): 3-D array containing the end orientation
+        samples (int): The number of samples used to interpolate
+    """
+    start_orientation = p.getEulerFromQuaternion(start_orientation)
+    end_orientation = p.getEulerFromQuaternion(end_orientation)
+    euler_orientations = np.linspace(start_orientation,
+                                     end_orientation, num=samples)
+    orientations = []
+
+    for euler_orientation in euler_orientations:
+        orientation_quat = p.getQuaternionFromEuler(euler_orientation)
+        orientations.append(orientation_quat)
+
+    return np.array(orientations).transpose()
+
+
+def spline_interpolation(points: np.array, samples: int, start_orientation: np.array = None,
+                         end_orientation: np.array = None):
     """Interpolates between a number of points in cartesian space.
 
     Args:
         points (np.array(3,n)): A 3 dimensional array whith each dimension containing
                                    subsequent positions.
         samples (int): The number of samples used to interpolate
+        start_orientation (np.array): Start orientation as quaternion
+        end_orientation (np.array): End orientation as quaternion
+
 
     Returns:
         ToolPath: A ToolPath object of the interpolated path
     """
     s = np.linspace(0, 1, len(points[0]))
 
-    path = np.zeros((3, samples))
+    positions = np.zeros((3, samples))
     cs_x = sci.CubicSpline(s, points[0])
     cs_y = sci.CubicSpline(s, points[1])
     cs_z = sci.CubicSpline(s, points[2])
 
     cs_s = np.linspace(0, 1, samples)
-    path[0] = cs_x(cs_s)
-    path[1] = cs_y(cs_s)
-    path[2] = cs_z(cs_s)
+    positions[0] = cs_x(cs_s)
+    positions[1] = cs_y(cs_s)
+    positions[2] = cs_z(cs_s)
+
+    final_path = ToolPath(positions=positions)
+
+    if start_orientation is not None and end_orientation is not None:

```

Comment:
This is unexpected behavior. If the positions are interpolated as splines so should the orientations.

This requires splines that can deal with the jump from 0 to -2pi!

Reply:
@liquidcronos @MaHajo in the new push of my code the spline interpolation was reset to the default code. There are issues that have been encountered using this method. For verification you can check the open issue: https://github.com/WBK-Robotics/pybullet_industrial/issues/50